### ISSUE TITLE
[Feature] Enable Local Chat

### ISSUE THREAD

**Issue by JoshsCode** [EarthMC/Issue-Tracker#196]
I think you should enable local chat to the chat channels. I know its already a feature and just needs to be enabled. I have come across towns near me and visited but found it awkward to chat with them because its hard to follow general chat and messaging can get spammy and annoying. It would also be good when meeting someone somewhere to trade with them.

**Comment by Ultil0l** [comment]
there is town chat, but what I think would be nice is a ranged chat like rp servers have

**Comment by KarlOfDuty** [comment]
does /local not work?

**Comment by Pengun78** [comment]
whats local chat

**Comment by Ultil0l** [comment]
^

**Comment by Arganog** [comment]
Interesting

How local is local?
Sub 1000 blocks? 

I can certainly see a use for this, for trying to talk with more than one person who you are near locally who aren't in your town/nation
(Party chat takes a while to set up, and /msg is just ideal for 1v1 private talk)

Preferably local chat would have a distinct chat color so it is easy to distinguish from general chat

(Donator Chat and Nation Chat both being yellow along with people leaving and joining the server being yellow makes yellow hard to see sometimes x.x)

**Comment by Jettermann** [comment]
Other servers tend to have it white. And it’s usually no more then a few hundred blocks like 250 maybe.

**Comment by Arganog** [comment]
Good and good, I like it

**Comment by Pengun78** [comment]
you could just do /tc or /nc?

**Comment by Jettermann** [comment]
Can you not understand what I wrote? It’s for if you are trading with some people in a town you aren’t allied with or you come across or live near a town you aren’t allied with then you could visit and talk to them all at the same time. Not everyone is an ally.

**Comment by Pengun78** [comment]
ah

**Comment by IGNSlaid** [comment]
we could have a local-only chat system, it would be very realistic. Also, the admins could have a way to overpass the range limitations and see all the conversations at once.
besides every local chat plugin has a configurable range...

about mr toxic mouth, a /ignore should solve the problem.

**Comment by Ultil0l** [comment]
that sounds nice

**Comment by JoshsCode** [comment]
I am not saying we should have a local only chat system but that would be more realistic and more interesting. Like we could have a donator chat, town chat, nation chat, and maybe trade chat that would all be at all range but then have a local chat also that would be like 500-1000 blocks maybe. It would get rid of the cancerous chat. But would change it up a lot (not saying we should just saying it would be interesting).

**Comment by IGNSlaid** [comment]
100 blocks would be fine, since nobody is trading within 6 chunks o' distance of eachother

Local chat would add alot of new mechanics to the game, since as meetings for strategical purposes and stuff like that, but yes, it would mean a big change to the way we play in the server.

**Comment by DKS134** [comment]
I'm totally up for this, it is impossible to talk to 3 players of different nations privatly about important shit without all the server knowing because the main chat is global

**Comment by Ultil0l** [comment]
go commit bad

**Comment by BowShot118** [comment]
I'm not against it, just stating alternatives.

**Comment by Jettermann** [comment]
Bowshot the point is if it’s not someone in your town or nation so chances are it won’t be easy to get them on discord or join a party with them for a 10 minute interaction. It is you are passing by a town and want to stop by and say hi to the people there and maybe do some trading.

**Comment by Jettermann** [comment]
And for the people who said it won’t be moderated we’ll messages apparently aren’t moderated either. I got spammed “Where the fuck did you run off to n****r?” Dozens of times but I had to make a report for it.

**Comment by BowShot118** [comment]
If it was via messages it makes sense. If you want a 1 on 1 convosation leave general chat and use /msg. It may be hard to juggle messages but you make it work. It just opens up opportunitys to spam without restriction due to the chat being unmoderated.

**Comment by Ultil0l** [comment]
also a local chat makes speaking easyer since you get less text pollution from more "vocal" players

**Comment by BowShot118** [comment]
You could still see General

**Comment by Coffeetom** [comment]
I would love to see this added. I have used it many times in other servers. Due to the large map size, I feel like a 500-1000 block area would be good. For the command, could it be set to /lc, so it is similar to other  chat commands. For tag in front of it, it could be a white "Local".

**Comment by SlyPrince** [comment]
"The main problem for me is that local chat is hard to monitor. Mods have to be in the chat range to see what is being said. We do not have enough mods to cover this map."

"It just opens up opportunitys to spam without restriction due to the chat being unmoderated."

Are you forgetting mods can't see town chat, nation chat and party chat already. It's not like local chat would be some loophole to be toxic and unmonitored by staff. 

Just as with tc, nc and pc, people do actually have the ability to use their brain and hands to submit a -new report of someone spamming n****r in local chat. They can also /ignore

**Comment by Pengun78** [comment]
Johnny doodoodoodoodoodoodoo johnny doodoodoodoodoodoodoo

**Comment by TypicalFernie** [comment]
They keep minimizing Sly's comments lol

**Comment by SlyPrince2** [comment]
Indeed, it's very interesting TypicalFernie.

Anyway as i was saying, it's no more difficult for mods to monitor as town chat is, and enabling it only adds to the EarthMC realism and immerrsion experience 🤷‍♂️

**Comment by KarlOfDuty** [comment]
Yeah probably.

**Comment by KarlOfDuty** [comment]
Done.
